dsp/modules/clarifai.py

```diff
@@ -0,0 +1,88 @@
+import math
+from typing import Any, Optional
+import backoff
+
+from dsp.modules.lm import LM
+
+try:
+    from clarifai.client.model import Model
+except ImportError:
+    raise ImportError("ClarifaiLLM requires `pip install clarifai`.")
+
+class ClarifaiLLM(LM):
+    """Integration to call models hosted in clarifai platform."""
+
+    def __init__(
+        self,
+        model: str = "https://clarifai.com/mistralai/completion/models/mistral-7B-Instruct", #defaults to mistral-7B-Instruct
+        api_key: Optional[str] = None,
+        stop_sequences: list[str] = [],
+        **kwargs,
+    ):
+        super().__init__(model)
+
+        self.provider = "clarifai"
+        self.pat=api_key
+        self._model= Model(url=model, pat=api_key)
+        self.kwargs = {
+            "n": 1,
+            **kwargs
+        }
+        self.history :list[dict[str, Any]] = []
+        self.kwargs['temperature'] = (
+            self.kwargs['inference_params']['temperature'] if
+            'inference_params' in self.kwargs and
+             'temperature' in self.kwargs['inference_params'] else 0.0
+        ) 
+        self.kwargs['max_tokens'] = (
+            self.kwargs['inference_params']['max_tokens'] if
+            'inference_params' in self.kwargs and
+             'max_tokens' in self.kwargs['inference_params'] else 150
+        )
```

Comment:
Not sure why this being done as in L46 only `self.kwargs['inference_params']` is used.

Reply:
Actually there's a constraint on parameters passing, since in our platform we support different models with different parameters, but in Dspy we can't necessarily leave the max_tokens and temp parameter in the class .
because internally it calls the predict function [refer](https://github.com/mogith-pn/dspy-fork/blob/8c6ba343d3fc67219f577e321605ba6accf292b0/dsp/primitives/predict.py#L102), which needs the default parameters of max_tokens and temperature, also 'n' value is used for number of generations in __call__ func. 
Some community models won't support these parameters, if we are passing as a whole from self.kwargs, so if the user wants to pass parameters using clarifaiLM they might want to declare it in inference params and if they don't it will consider the default value for prediction.

Reply:
Ah, I see